fedimint-api/src/db/mod.rs

```diff
@@ -680,6 +680,40 @@ where
     }
 }
 
+/// This is a helper macro that generates the implementations of
+/// DatabaseKeyPrefixConst necessary for reading/writing to the
+/// database and fetching by prefix.
+///
+/// - `key`: This is the type of struct that will be used as the key into the database.
+/// - `value`: This is the type of struct that will be used as the value into the database.
+/// - `prefix`: Enum expression that is represented as a `u8` that is the prefix prepended to this key.
```

Comment:
is there a reason why this is not `db_prefix` or `prefix_db` anymore?

Reply:
No reason. We can rename in a follow up if you'd like

Reply:
what do you think of the idea? 

IMHO, it may improve readability and reduce the time to understand how these prefixes work together. The `db_prefix` is always there as it is essential for the type. And then there usually is at least one `query_prefix` that returns all the values for a type.  (how is that enforced in the macro actually? is `$(, key_prefix = $prefix_ty:ty)* $(,)?` *zero or more* or *one or more*?)    

rename 

- `prefix` to `db_prefix`
- `key_prefix` to `query_prefix`

Reply:
> (how is that enforced in the macro actually? is $(, key_prefix = $prefix_ty:ty)* $(,)? zero or more or one or more?)

Yes exactly.

I'm good with all of the changes proposed here, feel free to open PR or I will get to it soon :)

Reply:
I can do it if you like :) depends on what soon is :'D